BABEL-3802 Bigint/Int/Smallint/Tinyint Power Function

pratikzode opened the pull request:
At first in Babel, the Transact-SQL functions power always return float instead of 
returning the same type in SQL server.

In this commit we have added Bigint, Int, Smallint and Tinyint to power function.
Have added the test case to show Bigint, Int, Smallint and Tinyint to power.

Task: BABEL-3802
Signed-off-by: pratikzode <pratik.zode@gmail.com>

macwoody commented on contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql:
```diff
@@ -49,6 +49,21 @@ CREATE OR REPLACE FUNCTION sys.degrees(IN arg1 TINYINT)
 RETURNS int AS 'babelfishpg_tsql','smallint_degrees' LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION sys.degrees(TINYINT) TO PUBLIC;
 
+CREATE OR REPLACE FUNCTION sys.power(IN arg1 BIGINT, IN arg2 FLOAT)
```
I think you may need to modify a different upgrade file in 3x.dev , please double check

macwoody commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -1570,3 +1574,51 @@ smallint_radians(PG_FUNCTION_ARGS)
+	Numeric	arg1_numeric, result_numeric;
+
+	arg1_numeric = DatumGetNumeric(DirectFunctionCall1(int8_numeric,arg1));
+	result_numeric = DatumGetNumeric(DirectFunctionCall2(numeric_power, NumericGetDatum(arg1_numeric),Int64GetDatum (arg2)));
```
for arg2, we still need to use NumericGetDatum

macwoody commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -1570,3 +1574,51 @@ smallint_radians(PG_FUNCTION_ARGS)
+	Numeric	arg1_numeric, result_numeric;
+
+	arg1_numeric = DatumGetNumeric(DirectFunctionCall1(int4_numeric,arg1));
+	result_numeric = DatumGetNumeric(DirectFunctionCall2(numeric_power, NumericGetDatum(arg1_numeric), Int32GetDatum (arg2)));
```
same here

macwoody commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -1570,3 +1574,51 @@ smallint_radians(PG_FUNCTION_ARGS)
+{
+	int16	arg1 = PG_GETARG_INT16(0);
+	Numeric	arg2 = PG_GETARG_NUMERIC(1);
+	int16	result;
```
I remember we return int type, isn't it?

pratikzode committed:
BABEL-3802 Bigint/Int/Smallint/Tinyint Power Function (#1123)

This Jira is a part of BABEL-736.
At first in Babel, the Transact-SQL functions power always return float instead of returning the same type in SQL server.
In this Jira we have added Bigint, Int, Smallint and Tinyint to power function.
Have added the test case to show Bigint, Int, Smallint and Tinyint to power.
Now this 3 is giving same output as SQL server.

Task: BABEL-3802
Signed-off-by: pratikzode <pratik.zode@gmail.com>

pratikzode committed:
Update schedule
